groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -0,0 +1,1571 @@
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_A >= 9
+
+#if BSLSTL_OPTIONAL_VARIADIC_LIMIT_A >= 10
+    template <class INIT_LIST_TYPE, class ARGS_01,
+                                    class ARGS_02,
+                                    class ARGS_03,
+                                    class ARGS_04,
+                                    class ARGS_05,
+                                    class ARGS_06,
+                                    class ARGS_07,
+                                    class ARGS_08,
+                                    class ARGS_09,
+                                    class ARGS_10>
+    void emplace(bslma::Allocator                           *allocator,
+                 std::initializer_list<INIT_LIST_TYPE>       initializer_list,
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_01),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_02),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_03),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_04),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_05),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_06),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_07),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_08),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_09),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_10));
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_A >= 10
+
+
+#endif
+#else
+// The generated code below is a workaround for the absence of perfect
+// forwarding in some compilers.
+    template <class... ARGS>
+    void emplace(bslma::Allocator                           *allocator,
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_GENERALIZED_INITIALIZERS)
+    template <class INIT_LIST_TYPE, class... ARGS>
+    void emplace(bslma::Allocator                           *allocator,
+                 std::initializer_list<INIT_LIST_TYPE>       initializer_list,
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+#endif
+// }}} END GENERATED CODE
+#endif
+
+    void reset() BSLS_KEYWORD_NOEXCEPT;
+        // Destroy the 'value_type' object in 'd_buffer', if any.
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS)
+    TYPE&  value() &;
+    TYPE&& value() &&;
+        // Return the 'value_type' object in 'd_buffer' with const
+        // qualification adjusted to match that of 'TYPE'.  The behavior is
+        // undefined unless 'this->hasValue() == true'
+#else
+    TYPE& value();
+        // Return the 'value_type' object in 'd_buffer' with const
+        // qualification adjusted to match that of 'TYPE'.  The behavior is
+        // undefined unless 'this->hasValue() == true'
+#endif  // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+    //ACCESSORS
+    bool hasValue() const BSLS_KEYWORD_NOEXCEPT;
+        // return 'true' if there is a value in 'd_buffer', and 'false'
```

Comment:
We try to avoid referring to imp details in contracts, e.g.:
```suggestion
        // Return 'true' if this objects has a value, and 'false'
```
Kinda tautological here, but, well, function name says it all pretty much.